compiler/rustc_mir/src/interpret/intrinsics.rs

```diff
@@ -337,6 +338,22 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let result = Scalar::from_uint(truncated_bits, layout.size);
                 self.write_scalar(result, dest)?;
             }
+            sym::const_allocate => {
```

Comment:
Thinking about this some more, I think this is the wrong place for this intrinsic. `interpret/` is for code that is shared among all machine instances, but this intrinsic is CTFE-only -- Miri does not want it (or it wants a different implementation).

The right folder for this code is `rustc_mir/src/const_eval`, somewhere in this `match`:

https://github.com/rust-lang/rust/blob/d37afad0cc87bf709ad10c85319296ac53030f03/compiler/rustc_mir/src/const_eval/machine.rs#L287

Reply:
done